Source/WebCore/editing/FrameSelection.cpp

```diff
@@ -2299,31 +2319,18 @@ void FrameSelection::setCaretVisibility(CaretVisibility visibility, ShouldUpdate
         updateSelectionAppearanceNow();
 
 #if ENABLE(TEXT_CARET)
-    if (m_caretPaint) {
-        m_caretPaint = false;
+    if (m_currentCaretAnimation->visible()) {
```

Comment:
Simpler to just call `m_currentCaretAnimation->setVisible(false)` unconditionally